argilla-server/src/argilla_server/jobs/dataset_jobs.py

```diff
@@ -0,0 +1,53 @@
+#  Copyright 2021-present, the Recognai S.L. team.
+#
+#  Licensed under the Apache License, Version 2.0 (the "License");
+#  you may not use this file except in compliance with the License.
+#  You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#  See the License for the specific language governing permissions and
+#  limitations under the License.
+
+from uuid import UUID
+
+from rq import Retry
+from rq.decorators import job
+
+from sqlalchemy import func, select
+
+from argilla_server.models import Record
+from argilla_server.database import AsyncSessionLocal
+from argilla_server.jobs.queues import default_queue
+from argilla_server.search_engine.base import SearchEngine
+from argilla_server.settings import settings
+from argilla_server.contexts import distribution
+
+JOB_TIMEOUT_DISABLED = -1
+JOB_RECORDS_YIELD_PER = 100
+
+
+@job(default_queue, timeout=JOB_TIMEOUT_DISABLED, retry=Retry(max=3))
+async def update_dataset_records_status_job(dataset_id: UUID):
+    """This Job updates the status of all the records in the dataset when the distribution strategy changes."""
+
+    record_ids = []
+
+    async with AsyncSessionLocal() as db:
+        stream = await db.stream(
+            select(Record.id)
+            .where(Record.dataset_id == dataset_id)
```

Comment:
Adding existing responses may reduce the number of records to evaluate.

```suggestion
            .where(Record.dataset_id == dataset_id)
            .join(Response)
```

Reply:
This could work because we're not right now deleting responses, but if in the future we delete responses for some reason this optimization will not update the status of records correctly.

Reply:
In those cases, the status computation doesn't need to be applied since the status is computed based on the record responses. Also, the number of min_submitted must be >= 1. 

We can review this optimization in the future if we support other acceptance criteria.

Reply:
Done.